libdnf/sack/advisory.cpp

```diff
@@ -0,0 +1,205 @@
+
+Advisory::~Advisory() = default;
+
+bool
+Advisory::operator==(const Advisory & other) 
+{
+    return sack == other.sack && advisory == other.advisory;
+}
+
+const char *
+Advisory::getDescription()
+{
+    return pool_lookup_str(dnf_sack_get_pool(sack), advisory, SOLVABLE_DESCRIPTION);
+}
+
+
+DnfAdvisoryKind
+Advisory::getKind()
+{
+    const char *type;
+    type = pool_lookup_str(dnf_sack_get_pool(sack), advisory, SOLVABLE_PATCHCATEGORY);
+    return str2dnf_advisory_kind(type);
+}
+
+const char *
+Advisory::getName()
+{
+    const char *name;
+
+    name = pool_lookup_str(dnf_sack_get_pool(sack), advisory, SOLVABLE_NAME);
+    size_t prefix_len = strlen(SOLVABLE_NAME_ADVISORY_PREFIX);
+    assert(strncmp(SOLVABLE_NAME_ADVISORY_PREFIX, name, prefix_len) == 0);
+    //remove the prefix
+    name += prefix_len;
+
+    return name;
+}
+
+void
+Advisory::getPackages(std::vector<AdvisoryPkg> & pkglist, bool withFilemanes)
+{
+    Dataiterator di;
+    const char * filename = nullptr;
+    Pool *pool = dnf_sack_get_pool(sack);
+
+    dataiterator_init(&di, pool, 0, advisory, UPDATE_COLLECTION, 0, 0);
+    while (dataiterator_step(&di)) {
+        dataiterator_setpos(&di);
+        Id name = pool_lookup_id(pool, SOLVID_POS, UPDATE_COLLECTION_NAME);
+        Id evr = pool_lookup_id(pool, SOLVID_POS, UPDATE_COLLECTION_EVR);
+        Id arch = pool_lookup_id(pool, SOLVID_POS, UPDATE_COLLECTION_ARCH);
+        if (withFilemanes) {
+            filename = pool_lookup_str(pool, SOLVID_POS, UPDATE_COLLECTION_FILENAME);
+        }
+        pkglist.emplace_back(sack, advisory, name, evr, arch, filename);
+    }
+    dataiterator_free(&di);
+}
+
+void
+Advisory::getReferences(std::vector<AdvisoryRef> & reflist)
+{
+    Dataiterator di;
+    Pool *pool = dnf_sack_get_pool(sack);
+
+    dataiterator_init(&di, pool, 0, advisory, UPDATE_REFERENCE, 0, 0);
+    for (int index = 0; dataiterator_step(&di); index++) {
+        reflist.emplace_back(sack, advisory, index);
+    }
+    dataiterator_free(&di);
+}
+
+const char *
+Advisory::getRights()
+{
+    return pool_lookup_str(dnf_sack_get_pool(sack), advisory, UPDATE_RIGHTS);
+}
+
+const char *
+Advisory::getSeverity()
+{
+    return pool_lookup_str(dnf_sack_get_pool(sack), advisory, UPDATE_SEVERITY);
+}
+
+const char *
+Advisory::getTitle()
+{
+    return pool_lookup_str(dnf_sack_get_pool(sack), advisory, SOLVABLE_SUMMARY);
+}
+
+unsigned long long int
+Advisory::getUpdated()
+{
+    return pool_lookup_num(dnf_sack_get_pool(sack), advisory, SOLVABLE_BUILDTIME, 0);
+}
+
+bool
+Advisory::matchBug(const char *bug)
+{
+    return matchBugOrCVE(bug, true);
+}
+
+bool
+Advisory::matchCVE(const char *cve)
+{
+    return matchBugOrCVE(cve, false);
+}
+
+bool
+Advisory::matchKind(const char *kind)
+{
+    return strcmp(
+        pool_lookup_str(dnf_sack_get_pool(sack), advisory, SOLVABLE_PATCHCATEGORY), kind) == 0;
+}
+
+
+bool
+Advisory::matchName(const char *name)
+{
+    
+    return strcmp(getName(), name) == 0;
```

Comment:
getName() can return nullptr. Do not pass nullptr into strcmp(). 

More safe code:
```
auto advisoryName = getName();
return advisoryName ? strcmp(advisoryName, name) == 0 : false;
```